examples/keras_spark_mnist.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import print_function
+
+import argparse
+import os
+import subprocess
+from distutils.version import LooseVersion
+
+import numpy as np
+
+import pyspark.sql.types as T
+from pyspark import SparkConf
+from pyspark.ml.evaluation import MulticlassClassificationEvaluator
+from pyspark.ml.feature import OneHotEncoderEstimator
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import udf
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras.models import Sequential
+from tensorflow.keras.layers import Dense, Dropout, Flatten
+from tensorflow.keras.layers import Conv2D, MaxPooling2D
+
+import horovod.spark.keras as hvd
+from horovod.spark.common.store import LocalStore, HDFSStore
+
+parser = argparse.ArgumentParser(description='Keras Spark MNIST Example',
+                                 formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+parser.add_argument('--master',
+                    help='spark master to connect to')
+parser.add_argument('--num-proc', type=int,
+                    help='number of worker processes for training, default: `spark.default.parallelism`')
+parser.add_argument('--batch-size', type=int, default=128,
+                    help='input batch size for training')
+parser.add_argument('--epochs', type=int, default=12,
+                    help='number of epochs to train')
+parser.add_argument('--work-dir', default='/tmp',
+                    help='temporary working directory to write intermediate files (prefix with hdfs:// to use HDFS)')
+
+args = parser.parse_args()
+
+# Initialize SparkSession
+conf = SparkConf().setAppName('keras_spark_mnist').set('spark.sql.shuffle.partitions', '16')
+if args.master:
+    conf.setMaster(args.master)
+elif args.num_proc:
+    conf.setMaster('local[{}]'.format(args.num_proc))
+spark = SparkSession.builder.config(conf=conf).getOrCreate()
+
+# Setup our store for intermediate data
+work_dir = args.work_dir
+hdfs_prefix = 'hdfs://'
+if work_dir.startswith(hdfs_prefix):
```

Comment:
This code could go into a class method of, lets say `Store.create(path)` that given a path returns an `HDFSStore` for a path starting with `hdfs://` and an `LocalStore` for a path starting with `file://`. This way, no-one needs to duplicate this 5 lines of code.

Reply:
I like it!  Thanks for the suggestion.

Reply:
I think this solution will work well for most open source users, who just want Store to mirror the behavior of PySpark when reading and writing files.  So I think it's good to simplify things behind this API.